blocks/procedures.js

```diff
@@ -435,13 +470,30 @@ Extensions.registerMixin(
     'procedure_def_update_shape_mixin', procedureDefUpdateShapeMixin);
 
 /** @this {Block} */
-const procedureDefGetLegalNameHelper = function() {
+const procedureDefValidatorHelper = function() {
   const nameField = this.getField('NAME');
   nameField.setValue(Procedures.findLegalName('', this));
-  nameField.setValidator(Procedures.rename);
+
+  /**
+   * Validates the input to the procedure name field.
+   * @param {string} newName The new name of the procedure.
+   * @return {string} The validated/legal name of the procedure.
+   * @this {Field}
+   */
+  const validator = function(newName) {
+    // TODO: Remove call the rename?
+    Procedures.rename.call(this, newName);
+
+    const sourceBlock = this.getSourceBlock();
+    const legalName = Procedures.findLegalName(newName.trim(), sourceBlock);
+    sourceBlock.model.setName(legalName);
+    return legalName;
```

Comment:
Good call! When I was writing this I wasn't sure if we wanted everything to be 100% backwards compatible or not. If not, I was planning on removing `rename` so I duplicated the logic. But given we do want it to be backwards compatible, this is unnecessary.

Skipped one test that was breaking because of the `isProcedureBlock` check I added. Will re-enable in the next PR once procedure definitions have a `doProcedureUpdate` method.